openedx/features/content_type_gating/models.py

```diff
@@ -57,36 +50,24 @@ class ContentTypeGatingConfig(StackedConfigurationModel):
     )
 
     @classmethod
-    def has_full_access_role_in_masquerade(cls, user, course_key, course_masquerade, student_masquerade,
-                                           user_partition):
+    def has_full_access_role_in_masquerade(cls, user, course_key, course_masquerade):
         """
         The roles of the masquerade user are used to determine whether the content gate displays.
         The gate will not appear if the masquerade user has any of the following roles:
         Staff, Instructor, Beta Tester, Forum Community TA, Forum Group Moderator, Forum Moderator, Forum Administrator
         """
-        if student_masquerade:
+        if is_masquerading_as_specific_student(user, course_key):
             # If a request is masquerading as a specific user, the user variable will represent the correct user.
             if user and user.id and has_staff_roles(user, course_key):
```

Comment:
Hmm. Maybe user and user.id are overly cautious. But it's definitely possible for user to not have staff roles, because `user` here is not the original user, but the student being masqueraded as. (i.e. we're checking if we're masquerading as a staff user)

Reply:
Good point, I didn't consider that case (staff masquerading as a specific user who _also_ happens to be staff).

And I only considered removing the `user` checks if we had been removing the `has_staff_roles` call (which I agree now, we shouldn't), so nothing to worry about here :)